cache.go

```diff
@@ -360,6 +426,10 @@ loop0:
 			}
 		}
 		idx := 0
+		// for some reason v2 sorting is backwards
```

Comment:
I guess if this remains like this we should change version property or add some other uniqueness to keys to avoid collisions with old keys (assuming old keys become visible in new API at some point).